src/OrchardCore.Modules/OrchardCore.Users/Migrations.cs

```diff
@@ -42,10 +42,33 @@ public int UpdateFrom3()
                 .AddColumn<bool>(nameof(UserIndex.IsEnabled), c => c.NotNull().WithDefault(true)));
 
             SchemaBuilder.AlterTable(nameof(UserIndex), table => table
-                .CreateIndex("IDX_UserIndex_IsEnabled", "IsEnabled")
+                .CreateIndex($"IDX_{nameof(UserIndex)}_{nameof(UserIndex.IsEnabled)}", nameof(UserIndex.IsEnabled))
             );
 
             return 4;
         }
+        public int UpdateFrom4()
+        {
+            SchemaBuilder.AlterTable(nameof(UserIndex), table =>
+            {
+                table.CreateIndex($"IDX_{nameof(UserIndex)}_{nameof(UserIndex.NormalizedUserName)}", nameof(UserIndex.NormalizedUserName));
+                table.CreateIndex($"IDX_{nameof(UserIndex)}_{nameof(UserIndex.NormalizedEmail)}", nameof(UserIndex.NormalizedEmail));
```

Comment:
I'm not sure but shouldn't UserName and Email be unique? Maybe it's just not enforced in the DB but it is in the code.

Reply:
So, as I see NormalizedUserName could be null for multiple users. And NormalizedEmail could also be null for multiple users. I think these 2 should be non-null and Unique.

Reply:
I think I agree @Skrypt

Reply:
Sorry but I don't get how your comment is related with the purpose of the PR. I'm adding indexes for those fields used in any query for making queries faster. But you are asking me to add constraints to the db? 
That should be matter of another PR,isn't it? 
In fact, IMO is better to add only those constraints you really need cause each extra constraint you add to the db hits a bit the performance (I've seen projects where no contraints where used because they hit performance a lot in comparison with not havind them).
But as I mentioned I think it should be another discursion in another PR. 
What do you think guys?

Reply:
Maybe I didn't unterstood sth

Reply:
The indexes added in this PR are fine if they are added for performance reason but While we are at it why not fixing the indexes constraints and else... I made a review of the code and found some issues in it and reported it. You can fix it with this PR or not as you wish.

Reply:
@jersiovic Normally your database should use constraints because changes could be coming from other applications using the same database tables. In fact here, the UserIndex should be unique and non-null to avoid having any scripts that could try to spoof a user or email by adding a record in these tables. This is an extra layer of security. In fact this UserIndex is more sensible than any other tables in OC so it makes sense to use database constraints on them to be sure that the data is compliant with our business logic.

Someone could say that if someone hacks this table that's because he got access to the database in the first place. But it is quite easy to run SQL Queries in OC from the Admin UI. It's worth nothing to test if someone could do a simple INSERT from these SQL Queries to see what I mean.